In [1]:
from cidereval import CiderD, Cider
from cidereval.tokenizer import SimpleTokenizer
from cidereval.tokenizer import PTBTokenizer

In [2]:
# demo script for running CIDEr
import time
import pdb
import json
from collections import defaultdict

from dataformat.loadData import LoadData
from cidereval.eval import CIDErEvalCap as ciderEval

pathToData = 'jsons/'

refName = 'pascal50S.json'
candName = 'pascal_candsB.json'

result_file = 'results.json'

In [3]:
# load reference and candidate sentences
loadDat = LoadData(pathToData)
gts, res = loadDat.readJson(refName, candName)

#res = res[:100]
#gts = {img['image_id']: gts[img['image_id']] for img in res}

In [4]:
tokenizer_old = PTBTokenizer('res')
_res_old = tokenizer_old.tokenize(res)

In [ ]:
tokenizer = SimpleTokenizer('res')
_res = tokenizer.tokenize(res)

for item, item_old in zip(_res, _res_old):
    if item['caption'][0] != item['caption'][0]:
        print(f"{item['caption'][0]} --> {item['caption'][0]}")

In [5]:

start_time = time.time()
tokenizer = SimpleTokenizer('gts')
_gts = tokenizer.tokenize(gts)

print("Spacy took: --- %s seconds ---" % (time.time() - start_time))

OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.

In [6]:
start_time = time.time()
tokenizer_old = PTBTokenizer('gts')
_gts_old = tokenizer_old.tokenize(gts)

print("PTBTokenizer took: --- %s seconds ---" % (time.time() - start_time))

PTBTokenizer took: --- 1.0661442279815674 seconds ---


for k in gts:
    for cap, cap_old in zip(_gts[k], _gts_old[k]):
        if cap != cap_old:
            print(f"{cap} -> {cap_old}")
        

In [7]:
scorer = Cider(df='coco-val')
scorerD = CiderD(df='coco-val')

In [8]:
score, scores = scorer.compute_score(_gts_old, _res_old)
scoreD, scoresD = scorerD.compute_score(_gts_old, _res_old)
print(score)
print(scoreD)

0.5580180601041953


In [10]:
scorer.save_df(df_name="prova")
scorerD.save_df(df_name="prova")

the path provided is not valid. The df will be saved in {path}
saved to /home/michele/prova.p
the path provided is not valid. The df will be saved in {path}
saved to /home/michele/prova.p


##  CiderD

### df in corpus mode 

corpus: 
- refName = 'pascal50S.json'
- candName = 'pascal_candsB.json'

spacy tokenizer : 0.48387229928700376

PTB tokenizer: 0.48439138950913807
    
**diff: 0.0005190902221343108**

### df in coco-val mode 

PTB tokenizer = 0.4594128636973375

spacy tokenizer = 0.45890712727870175


**diff = 0.0005057364186357716**

##  Cider

### df in corpus mode 

corpus: 
- refName = 'pascal50S.json'
- candName = 'pascal_candsB.json'

spacy tokenizer : 0.5870925361541144
PTB tokenizer: 0.5876485026978251
    
**diff: 0.0005559665437107064**

### df in coco-val mode 

PTB tokenizer = 0.5580180601041953

spacy tokenizer = 0.557550429560831

abs(0.557550429560831-0.5580180601041953) 

**diff = 0.00046763054336429466**

PTB Tokenizer does not tokenize :
 
 - word/word (slahes)
 - (word) (brackets) it returns something like -lrb-
 
 Spacy does it